### PR TITLE
Better error message for localizer, simpler error message.

### LINKED ISSUE #316: Brainomics URLs are scary
We should maybe hide the RQL request in the URL because it is a bit scary for non brainomics users.

### PR DESCRIPTION
**Pull request by AlexandreAbraham**
This PR add an error message if a string instead of a list of string is given in the localizer fetching function (I think that it may be a common mistake and the previous error message was cryptic). I also removed the variables for the URL when using default verbose option (verbose=0). The full URL is still displayed if verbose > 0 (fix #316).

### CODE REVIEW

File: nilearn/datasets.py [hunk 1]
```diff
@@ -1807,6 +1811,9 @@ def fetch_localizer_contrasts(contrasts, n_subjects=None, get_tmaps=False,
         BMC neuroscience 8.1 (2007): 91.
 
     """
+    if isinstance(contrasts, basestring):
+        raise ValueError('Constrasts should be a list of string, a single '
+                'string was given: "%s"' % contrasts)
```
Comment (lesteve) [inline nilearn/datasets.py]: PEP8

File: nilearn/datasets.py [hunk 2]
```diff
@@ -427,7 +427,11 @@ def _fetch_file(url, data_dir, resume=True, overwrite=False,
     initial_size = 0
     try:
         # Download data
-        print 'Downloading data from %s ...' % url
+        if verbose == 0:
+            # Remove variables from url for better display
+            print 'Downloading data from %s ...' % url.split('?')[0]
```
Comment (lesteve) [inline nilearn/datasets.py]: urllib.splitquery may be a bit clearer. It returns a tuple `url_without_query, query`

Also slight niggle, I would do 
```python

displayed_url = urllib.splitquery(url)[0] if verbose == 0 else url
print 'Dowloading data from %s ...' % displayed_url
```

**Comment by lesteve** [comment]
Just as a reminder to ourselves, in we ever forget about this, the verbose parameter need to be passed into _fetch_file.

**Comment by AlexandreAbraham** [comment]
Fixed in the last commit.

**Comment by bthirion** [comment]
LGTM

**Pull request merged**